test/common/network/connection_impl_test.cc

```diff
@@ -100,10 +100,17 @@ class ConnectionImplTest : public testing::TestWithParam<Address::IpVersion> {
     dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
   }
 
-  void disconnect() {
+  void disconnect(bool async) {
```

Comment:
This is really condition on whether we expect a remote close or not, rather than async?